### ISSUE TITLE
Method used for highlighting breaks with colors set by theme

### ISSUE THREAD

**Issue by JosefLitos** [rcarriga/nvim-dap-ui#178]
When I set the colors of `control_hl_groups` in my theme, it breaks upon switching to dark/light variant, where I update only the black/white colors. It seems that after setting these groups they all get cleared, while the groups like `DapUIScope` and all other outside `control_hl_groups` stay fine:
```
DapUIStop      xxx cleared
DapUIRestart   xxx cleared
DapUIStepOver  xxx cleared
DapUIStepInto  xxx cleared
DapUIStepOut   xxx cleared
DapUIStepBack  xxx cleared
DapUIPlayPause xxx cleared
DapUIUnavailable xxx cleared
DapUIThread    xxx links to Fg
```
This causes an error in `config/highlights.lua:74` since you are trying to format `nil`.
Also there is no reason to use vim.cmd for highlighting when 0.8 features setting the groups aside of just retrieving them.
I will try to rewrite all highlight groups to lua and see if that solves the issue. If I can get it solved that way, I would like to issue a PR.

**Comment by JosefLitos** [comment]
Also I don't think there is a need to set colors multiple times, because no colorscheme clears groups set by itself, so if the groups were set, they will stay so anyway. It would be better to set all groups on setup and then dont touch them again. Also, is there a particular reason why the value of control groups is copied instead of linking to the group? Do the groups have different use case from non-NC counterparts?

**Comment by JosefLitos** [comment]
The colors are being cleared by `patch_background()`.

**Comment by JosefLitos** [comment]
I understand you're trying to synchronize the background color of `WinBar` with all icons, but if no background is used, it should automatically be the color of background. that is, if the `WinBar` group highlight covers the entire line and on top are highlighted the separate icons.

**Comment by JosefLitos** [comment]
Or, if the non-NC groups are just unused templates, their background shouldn't be changed, since the group can be set by the colorscheme and this would modify the group.

**Comment by nyngwang** [comment]
I sincerely don't think what you did in #180 is correct.

**Comment by JosefLitos** [comment]
I am not sure I understand what you mean. What exactly do you mean by not being picked? What sets of values are being used, if they are different from the defaults, where do they come from, if not the colorscheme?

**Comment by JosefLitos** [comment]
The problem might be I still don't understand the philosophy behind `...NC` groups.

**Comment by nyngwang** [comment]
@JosefLitos Hi, first of all, thanks for your kind explanation.

> the philosophy behind `...NC` groups.

From my current observation, `...NC` groups are designed to have the effect like this plugin does: https://github.com/sunjon/Shade.nvim.

> What exactly do you mean by not being picked? [...] What sets of values are being used, if they are different from the defaults, where do they come from, if not the colorscheme?

I meant: (I tried)

1. Even if I have added `DapUINormal` in my colorscheme, it is either `cleared` or set to some strange value from nowhere (I tried `:verb hi DapUINormal` to find the source but apparently it's not useful.)
2. I have to add `vim.cmd('hi DapUINormal')` right before the command `dapui.open({ reset=true })` to make it take effect. So there must be some plugin(or maybe it's nvim-dap-ui itself) defining the same highlight group again in the middle.
